pgx/src/callbacks.rs

```diff
@@ -191,7 +192,7 @@ where
                 // effectively 'take' the hook from the internal RefCell
                 if let Some(hook) = hook.replace(None) {
                     // and execute it under guard for proper panic/elog(ERROR) handling
-                    crate::guard::guard(hook.0);
+                    pg_sys::guard(hook.0);
```

Comment:
This is an `extern "C"` fn so it does need to guard, but I think the guard should be over the entire callback, e.g. starting on line 163 and going to 198.

It looks like that's what `#[pg_guard]` does... but it also [applies `#[no_mangle]`](https://github.com/tcdi/pgx/blob/077c447543e863e67f2d774be4dfbf05372b26d9/pgx-utils/src/rewriter.rs#L72), which is super duper wrong for something like this!!

Reply:
Yup.  I've fixed up the no_mangle issues as well